### DISCUSSION TITLE
Dry-Run mode fails with gfal

### DISCUSSION THREAD

**Discussion by flgomezc** [dmwm/CMSRucio#699] [General]
This couple of lines fail when triying to do a Dry-Run 

https://github.com/dmwm/CMSRucio/blob/551fe134b151c4cae882ffbb1c3ee9e098fba6a0/UserDMTools/user_data_registration/src/utils/register_files.py#L23-L24

On dry-run the files are not yet on destination, but the code uses size and adler32 from destination.

**Comment by dynamic-entropy** [comment]
This is by design. Dry run for each script is applicable to only individual steps that the module takes care of and not the whole process of transferring data and registering. 
So `-dry-run` here  would test if all files are in place and ready to start the actual registration. Helping us to check the files.
